src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs

```diff
@@ -134,6 +134,18 @@ internal sealed partial class ConsoleHost
             }
 #endif
 
+            // put PSHOME in front of PATH so that calling `powershell` within `powershell` always starts the same running version
+            StringBuilder path = new StringBuilder(Environment.GetEnvironmentVariable("PATH"));
+            string pathSeparator = ";";
+            if (!Platform.IsWindows)
+            {
+                pathSeparator = ":";
+            }
+            string pshome = Utils.DefaultPowerShellAppBase;
+            path.Replace(pathSeparator + pshome, "");
+            path.Insert(0, pshome + pathSeparator);
```

Comment:
We should check if the first path in `$env:PATH` is `pshome`, and only insert if it's not. Otherwise, the nested powershell instance will have 2 `pshome` paths in `$env:PATH`